### DISCUSSION TITLE
Collections not loading into memory

### DISCUSSION THREAD

**Discussion by KevinKoreth** [milvus-io/milvus#37164] [Q&A and General discussion]
unable to load collections into memory

    Identified cause:

 

From query coordinator logs:
[INFO] [task/scheduler.go:269] ["task added"] [task="[id=1726294031046] [type=Grow] [source=1] [reason=lacks of segment] [collectionID=445296266907850607] [replicaID=445296481999638557] [priority=Normal] [actionsCount=1] [actions={[type=Grow][node=61][streaming=false]}] [segmentID=445296266748501819]"]

 

This suggests that the tasks are of type "Grow," which typically indicates an operation to expand or increase resources, such as adding more segments or nodes. The repeated mention of node 61 suggests that this node is heavily involved in the growth tasks.
![unnamed](https://github.com/user-attachments/assets/e09696f3-21ed-44c3-a751-993393598b1a)
The conditions indicate that the node's status is unknown (NodeStatusUnknown) for MemoryPressure, DiskPressure and PIDPressure.

This suggests that the Kubelet on the node has stopped posting status update
I want to know what is the best way to debug this currently all the collections are in not load state and pymilvus client times out while trying to load the collection

**Comment by yhmo** [comment]
[INFO] [task/scheduler.go:269] ["task added"] is an **INFO**, which is a normal log, not a problem.
The type "Grow" means this segment is a growing segment and keeps in memory of the "shard leader" query node.

This log has nothing to do with k8s. It is just an internal logic of milvus query node to manage the segments.
The collection is already in load state, it just tries to load all the segments into memory. 
If the client time out for load, the possible reasons:
- the query node's memory capacity is insufficient
- the data is huge, memory is sufficient, query nodes take a long time to load the data
- the s3 connection is broken

**Comment by xiaofan-luan** [comment]
> ```
> unable to load collections into memory
> 
> Identified cause:
> ```
> 
> From query coordinator logs: [INFO] [task/scheduler.go:269] ["task added"] [task="[id=1726294031046] [type=Grow] [source=1] [reason=lacks of segment] [collectionID=445296266907850607] [replicaID=445296481999638557] [priority=Normal] [actionsCount=1] [actions={[type=Grow][node=61][streaming=false]}] [segmentID=445296266748501819]"]
> 
> This suggests that the tasks are of type "Grow," which typically indicates an operation to expand or increase resources, such as adding more segments or nodes. The repeated mention of node 61 suggests that this node is heavily involved in the growth tasks. ![unnamed](https://private-user-images.githubusercontent.com/73425943/380153197-e09696f3-21ed-44c3-a751-993393598b1a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAxNDc0NTIsIm5iZiI6MTczMDE0NzE1MiwicGF0aCI6Ii83MzQyNTk0My8zODAxNTMxOTctZTA5Njk2ZjMtMjFlZC00NGMzLWE3NTEtOTkzMzkzNTk4YjFhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI4VDIwMjU1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExMGY3M2Y0OWMxY2EwZGIzYjFlN2UxMjIyYWQyZDE0YmQ2MTFlMGE1NWY0OWM1OGFiMGVlN2M2ZWVlMmFlNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3vHVkZIyGkK5JXI3zXrVRqzIKWL9LQma5E249IVYv88) The conditions indicate that the node's status is unknown (NodeStatusUnknown) for MemoryPressure, DiskPressure and PIDPressure.
> 
> This suggests that the Kubelet on the node has stopped posting status update I want to know what is the best way to debug this currently all the collections are in not load state and pymilvus client times out while trying to load the collection

Could you please refer this [doc](https://github.com/milvus-io/milvus/tree/master/deployments/export-log) to export the whole Milvus logs for investigation? For Milvus installed with docker-compose, you can use docker-compose logs > milvus.log to export

**Reply by KevinKoreth** [reply]
Thankyou for responding. 
This document says you need the name of the milvus instance, I am not sure what that means is it the pod name? If it's a pod then which pod?

**Reply by yhmo** [reply]
I suppose you deployed milvus with helm: https://milvus.io/docs/install_cluster-helm.md#Run-Milvus-in-Kubernetes-with-Helm
The name is defined when you launch the cluster, which is required by the export-milvus-log.sh
`helm install my-release milvus/milvus`
